src/app/qgsattributetabledialog.cpp

```diff
@@ -925,6 +929,8 @@ void QgsAttributeTableDialog::mActionRemoveAttribute_triggered()
     if ( mLayer->deleteAttributes( attributes ) )
     {
       mLayer->endEditCommand();
+
+      mLayer->setDisplayExpression( mLayer->fields().count() > 0 ? mLayer->fields().at( 0 ).name() : QString() );
```

Comment:
This sounds wrong to reset the display expression on any attribute removal. I think you would need to test if the fields removed were used in the expression before resetting it.
I would let it as is except if it was exactly one of the removed field name.

Reply:
Updated to reset display expression only if it is a single-field expression and this field was removed. In all other cases it is up to user to adjust display expression.

Reply:
perfect, well done.